neurodamus/io/synapse_reader.py

```diff
@@ -232,7 +232,13 @@ class SonataReader(SynapseReader):
     }
 
     def _open_file(self, src, population, _):
-        storage = libsonata.EdgeStorage(src)
+        try:
+            from mpi4py import MPI
+            hdf5_reader = libsonata.make_collective_reader(MPI.COMM_WORLD, False, True)
```

Comment:
if `mpi4py` is installed for other reasons, but the user doesn't have libsonata_mpi, then either we have an unhandled AttributeError or make_collective_reader will return something non-collective. Am I missing something?

Reply:
You avoid the former by requiring `libsonata>=0.1.25` (to be released) and the latter is intended because it prevents a hard dependency on MPI.

Reply:
I've just manually launched https://bbpgitlab.epfl.ch/hpc/sim/blueconfigs/-/pipelines/184571 with blueconfigs branch `weji/libsonata_mpi` and this neurodamus branch. The CI tests failed. Does it also require libsonata>=0.1.25 to be released ?

Reply:
That looks like a deadlock, most like due to something being non-collective.

We "released" a compatible version of `libsonata` on BB5, so I don't think that's the issue. If it were we'd expect something related to not being able to import `libsonata.make_collective_reader`.

Reply:
It's caused by this:
https://github.com/BlueBrain/HighFive/issues/911